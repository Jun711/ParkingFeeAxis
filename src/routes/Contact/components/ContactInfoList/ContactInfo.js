export default ContactInfo = [
  {
    "icon": "envelope-o",
    "method": "email",
    "text": "Please email me at juny.g711@gmail.com."
  },
  {
    "icon": "globe",
    "method": "web",
    "text": "Check out my website at www.language-diary.com"
  },
  {
    "icon": "twitter",
    "method": "twitter",
    "text": "Follow me at @Juny_g711"
  }
]
