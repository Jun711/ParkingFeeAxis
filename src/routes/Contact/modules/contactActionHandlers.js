import update from "immutability-helper/index"

const errorMsg = [{
  '_id': 'errorMsg',
  'icon': 'warning',
  'method': 'error',
  'text': 'Error: Contact info loading timeout'
}]

export function handleLoadContactInfo(state, action) {
  return update(state, {
    contactInfoLoaded: {
      $set: true
    },
    contactInfo: {
      $set: action.payload
    }
  })
}

export function handleLoadContactInfoError(state) {
  // TODO can make a global error page
  return update(state, {
    contactInfoLoaded: {
      $set: true
    },
    contactInfo: {
      $set: errorMsg
    }
  })
}