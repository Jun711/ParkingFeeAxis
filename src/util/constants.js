// config
export const LATITUDE_DELTA = 0.005
export const defaultAjaxTimeout = 30000
export const DEFAULT_LATITUDE = 49.2820
export const DEFAULT_LONGITUDE = -123.1171

// location permission constants
export const PERMISSION_REQ_TITLE = 'Parking App Location Permission'
export const PERMISSION_REQ_MSG = 'Parking App needs access to your location to locate nearby parking.'

// toast
export const TOAST_HEADER_Y_OFFSET = 145
export const TOAST_HEADER_X_OFFSET = 0
export const LOCATION_ERROR_MSG_ONE = 'Please enable location service.'
export const LOCATION_ERROR_MSG_TWO = 'Location service is unavailable.'
export const LOCATION_ERROR_DEFAULT_MSG = 'Location request timeout.'

// about
export const APP_NAME = 'Path'
export const APP_VERSION = 'Version 1.0.0'

// time
export const WEEKDAY_9_6 = 'weekdayOffice'
export const WEEKDAY_6_PLUS = 'weekdayAfterOffice'
export const SAT_9_6 = 'satOffice'
export const SAT_6_PLUS = 'satAfterOffice'
export const SUN_9_6 = 'sunOffice'
export const SUN_6_PLUS = 'sunAfterOffice'
export const FREE_PARKING = 'freeParking'
export const FREE_PARKING_LIMIT = 'until 9am'
export const NO_TIME_LIMIT_TEXT = 'no limit'


// min and max rate
export const MIN_PARKING_RATE = 100
export const MAX_PARKING_RATE = -1

// color
export const LOADER_COLOR = '#515c6d'
export const THEME_COLOR = '#38414e'
export const TEXT_COLOR = '#000'
// export const CENTRE_MARKER_COLOR = '#8585ad'
export const CURRENT_MARKER_COLOR = '#98a3b3'
export const CENTRE_MARKER_COLOR ='#515c6d'
export const WHITE_COLOR = '#fff'

// font size
export const TITLE_FONT_SIZE = 20
export const PRIMARY_FONT_SIZE = 18
export const SECONDARY_FONT_SIZE = 14
export const APP_NAME_FONT_SIZE = 15
export const APP_VERSION_FONT_SIZE = 10

// search bar
export const SEARCH_PLACEHOLDER = 'Search'
export const SEARCH_INPUT_KEY = 'headerSearchBar'

// error message
export const CONTACT_INFO_LOADING_ERROR = 'Contact info loading timeout'
export const FAQ_LOADING_ERROR = 'FAQ list loading timeout'
