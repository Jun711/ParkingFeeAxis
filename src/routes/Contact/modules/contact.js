// contains actions and handlers
import constants from './actionConstants'
import timeout from '../../../util/timeout'
import {
  CONTACT_INFO,
} from '../../../util/constants'
import * as actionHandlers from './contactActionHandlers'

//-------------------------------
// Utility function
//-------------------------------


//-------------------------------
// Actions
//-------------------------------
export function loadContactInfo() {
  return async (dispatch) => {
    try {
      const response = await timeout(fetch(CONTACT_INFO))
      const responseJson = await response.json()
      dispatch({
        type: constants.LOAD_CONTACT_INFO,
        payload: responseJson
      })
    } catch (error) {
      dispatch({
        type: constants.LOAD_CONTACT_INFO_ERROR,
        payload: error
      })
    }
  }
}

//-------------------------------
//Action Handlers
//-------------------------------
const ACTION_HANDLERS = {
  LOAD_CONTACT_INFO: actionHandlers.handleLoadContactInfo,
  LOAD_CONTACT_INFO_ERROR: actionHandlers.handleLoadContactInfoError,
}

//-------------------------------
// Initialization
//-------------------------------
export const initialState = {
  contactInfoLoaded: false,
  contactInfo: [],
  loadingError: false,
}

export function ContactReducer(state = initialState, action) {
  const handler = ACTION_HANDLERS[action.type]

  return handler ? handler(state, action) : state
}