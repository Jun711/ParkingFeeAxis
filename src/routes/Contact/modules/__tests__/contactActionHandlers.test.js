import { handleLoadContactInfo } from '../contactActionHandlers'
import { initialState } from '../contact'
import constants from '../actionConstants'
import ContactInfo from '../../components/ContactInfoList/ContactInfo'

describe(`handle ${constants.LOAD_CONTACT_INFO} action`, () => {
  it('returns the same state on an unhandled action', () => {
    expect(handleLoadContactInfo(initialState, {type: undefined})).toMatchSnapshot()
  })

  it(`handles ${constants.LOAD_CONTACT_INFO} success action`, () => {
    expect(handleLoadContactInfo(initialState, {type: constants.LOAD_CONTACT_INFO, payload: ContactInfo})).toEqual({
      ...initialState,
      contactInfoLoaded: true,
      contactInfo: ContactInfo
    })

    expect(handleLoadContactInfo(initialState, {
      type: constants.LOAD_CONTACT_INFO,
      payload: ContactInfo
    })).toMatchSnapshot()
  })
})


describe(`handle ${constants.LOAD_CONTACT_INFO_ERROR} action`, () => {
  it(`handles ${constants.LOAD_CONTACT_INFO_ERROR} action`, () => {
    const error = new Error('error')

    const errorMsg = [{
      '_id': 'errorMsg',
      'icon': 'warning',
      'method': 'error',
      'text': 'Error: Contact info loading timeout'
    }];

    expect(handleLoadContactInfo(initialState, {type: constants.LOAD_CONTACT_INFO, payload: errorMsg})).toEqual({
      ...initialState,
      contactInfoLoaded: true,
      contactInfo: errorMsg
    })

    expect(handleLoadContactInfo(initialState, {type: constants.LOAD_CONTACT_INFO_ERROR, error})).toMatchSnapshot()
  })
})