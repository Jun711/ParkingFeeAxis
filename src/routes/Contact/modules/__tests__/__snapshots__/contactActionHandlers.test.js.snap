// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle LOAD_CONTACT_INFO action handles LOAD_CONTACT_INFO success action 1`] = `
Object {
  "contactInfo": Array [
    Object {
      "icon": "envelope-o",
      "method": "email",
      "text": "Please email me at juny.g711@gmail.com.",
    },
    Object {
      "icon": "globe",
      "method": "web",
      "text": "Check out my website at www.language-diary.com",
    },
    Object {
      "icon": "twitter",
      "method": "twitter",
      "text": "Follow me at @Juny_g711",
    },
  ],
  "contactInfoLoaded": true,
}
`;

exports[`handle LOAD_CONTACT_INFO action returns the same state on an unhandled action 1`] = `
Object {
  "contactInfo": undefined,
  "contactInfoLoaded": true,
}
`;

exports[`handle LOAD_CONTACT_INFO_ERROR action handles LOAD_CONTACT_INFO_ERROR action 1`] = `
Object {
  "contactInfo": undefined,
  "contactInfoLoaded": true,
}
`;
